lib/std/crypto/ff.zig

```diff
@@ -835,20 +834,16 @@ const ct_protected = struct {
 
     // Compares two big integers in constant time, returning true if x < y.
     fn limbsCmpLt(x: anytype, y: @TypeOf(x)) bool {
-        assert(x.limbs_count() == y.limbs_count());
-        const x_limbs = x.limbs.constSlice();
-        const y_limbs = y.limbs.constSlice();
-
         var c: u1 = 0;
-        for (0..x.limbs_count()) |i| {
-            c = @as(u1, @truncate((x_limbs[i] -% y_limbs[i] -% c) >> t_bits));
+        for (x.limbsConst(), y.limbsConst()) |x_limb, y_limb| {
+            c = @truncate((x_limb -% y_limb -% c) >> t_bits);
         }
-        return @as(bool, @bitCast(c));
+        return c != 0;
     }
 
     // Compares two big integers in constant time, returning true if x >= y.
     fn limbsCmpGeq(x: anytype, y: @TypeOf(x)) bool {
-        return @as(bool, @bitCast(1 - @intFromBool(ct.limbsCmpLt(x, y))));
+        return !ct.limbsCmpLt(x, y);
```

Comment:
Should be fine on targets that have instructions like `SETx` to set registers according to flags, but doesn't that increase the risk of having conditional jumps being emitted? What do our own backends do?

Reply:
There is no such thing as "best effort" constant time. Without the language encoding constant time semantics, these two expressions are equivalent.

The issue for this is #1776.